ts/parts-map/HeatmapSeries.ts

```diff
@@ -456,6 +799,42 @@ seriesType<Highcharts.HeatmapSeries>(
 
         /* eslint-disable valid-jsdoc */
 
+        /**
+         * @private
+         * @function Highcharts.Point#applyOptions
+         * @param {Highcharts.HeatmapPointOptions} options
+         * @param {number} x
+         * @return {Highcharts.SVGPathArray}
+         */
+        applyOptions: function (
+            this: Highcharts.HeatmapPoint,
+            options: Highcharts.HeatmapPointOptions,
+            x?: number
+        ): Highcharts.HeatmapPoint {
+            var point = H.Point.prototype
+                .applyOptions.call(this, options, x) as any;
+
+            point.formatPrefix =
+                point.isNull || point.value === null ?
+                    'null' : 'point';
+
+            return point;
+        },
+        /**
+         * Color points have a value option that determines whether or not it is
+         * a null point
+         * @private
+         * @function Highcharts.HeatmapPoint.isValid
+         * @return {boolean}
+         */
+        isValid: function (this: Highcharts.HeatmapPoint): boolean {
+            // undefined is allowed
+            return (
+                this.value !== Infinity &&
+                this.value !== -Infinity
```

Comment:
This function should also check for `this.value !== null`, then we can use `point.isNull` safely above.